### ISSUE TITLE
error highlighting on inline comments

### ISSUE THREAD

**Issue by idiotWu** [daaain/Handlebars#60]
In ghost themes' [page templates](http://themes.ghost.org/v0.6.4/docs/templates#section-default-hbs), there is a `{{!< default}}` comment to specify that they extend the default template. It seems not to be treated as a comment with this plugin in sublime:

![screenshot](https://cloud.githubusercontent.com/assets/6022672/7764389/2a09f8aa-0083-11e5-8c0c-3fd7393ddbd6.jpg)

**Comment by idiotWu** [comment]
I've taken a look at the source code here: [Handlebars.JSON-tmLanguage#L374](https://github.com/daaain/Handlebars/blob/master/Handlebars.JSON-tmLanguage#L374), I just can't understand why if there's a comment that has `<` right after `{{!` begin tag should be exclude from comment?

**Comment by daaain** [comment]
No idea why that exclusion was there, thanks a lot for looking into it!

**Comment by hlvnst** [comment]
Bump!

I put that in there right here: 3cffad33f7bd7ba6fe2db4d1285de2cbca67b564. My reasoning was that (at least in Ghost) it's more like a directive than a comment. A comment is ignored, whereas Ghost treats `{{!< default}}` specially. I find it clearer to emphasise that special treatment than to display it as a regular comment.

Ghost documents the "comment" here: http://themes.ghost.org/docs/templates#section-default-hbs

**Comment by daaain** [comment]
Oh, sorry @hlvnst, should have had a better look!

And I agree with the reasoning that this should be treated specially, what do you think @idiotWu?

**Comment by idiotWu** [comment]
I'm afraid I can't agree with @hlvnst.

`{{! ... }}` is defined as comment blocks in Handlebars, but Ghost treats `{{!< ... }}` as a directive. In other word, **Ghost syntax is a super set of Handlebars**. 

So if you want to treat them separately, there should be two plugins, **one for Handlebars and another for Ghost**, like Javascript for default js files and Javascript Next for es6.

![screenshot](https://cloud.githubusercontent.com/assets/6022672/11719445/cbfc55d4-9f95-11e5-8524-4a4587f348d4.png)

**Comment by daaain** [comment]
Hmm, but why would you need a separate package for vanilla Javascript when ES6 covers that too? In my view the built in / vanilla JS package is just obsolete at this point and the best would be to disable it and use the more featured ES6 / Babel version which is also backwards compatible.

Or turned upside down in this case, since it's trivial to support the Ghost flavour in this Handlebars package, why would you fork this and maintain a Ghost package just for this one difference?

**Comment by hlvnst** [comment]
I agree with @daaain, especially on his last point. The difference is too minor to maintain a separate fork just for it. It's not like it breaks anything else.

**Comment by idiotWu** [comment]
Yep that what confusing me. Right now it seems to be only a slight difference between them, but what if Ghost adds more features in the future?

The first time I saw the highlighted `{{!< default }}`, I thought it was a special sentence and I missed it from Handlebars Documents. However I looked up whole documents time after time I just couldn't find any definition about it! Finally I googled it and put up this issue.

Well I was misled due to I was a newbee, but what's your feeling of facing highlighted `{{!I}}` `{{!A}}` `{{!M}}` `{{!G}}` `{{!H}}` `{{!O}}` `{{!S}}` `{{!T}}` ... okay I'm just joking.

**Comment by daaain** [comment]
Maybe the solution would be to treat this as a special case and tokenise differently so it gets a distinct colour?

Not sure yet what token type would I use though, just thinking out loud :)

**Comment by daaain** [comment]
OK, so I had a little bit of a dig and think, how about making this a `meta.preprocessor` type, which would get highlighted with the same distinct colour as numbers / escaped quotes? See: https://github.com/jonschlinkert/sublime-monokai-extended/blob/master/Monokai%20Extended.JSON-tmTheme#L56

**Comment by hlvnst** [comment]
Seems like a good idea. I'll see if I can get that going one of these days.

**Comment by hlvnst** [comment]
Or maybe not such a good idea.

I have no problem getting `{{!< default}}` recognized as a preprocessor statement, but it seems many (most?) themes don't have directives for `meta.preprocessor`, so then it doesn't get any highlighting at all.

**Comment by daaain** [comment]
Ah, shame :( I tried to choose a semantically relevant tag, but maybe we could then just force the colouring by adding `constant.numeric` too or some other widely supported tag with the same colour.

**Comment by hlvnst** [comment]
I'm not above that, because pragmatism. 😏

I've got the whole pattern recognized as `meta.preprocessor` so at least there's some relevant semantics there, but then I just used some random types for the captures to get the thing highlighted in a distinct way.

In my TextMate bundle, using the Twilight theme, this is what it looks like:

![screen shot 2016-01-21 at 3 00 10](https://cloud.githubusercontent.com/assets/696169/12469319/20364324-bfeb-11e5-8286-abe8a50e1150.png)

A pull request will follow shortly.

**Comment by daaain** [comment]
With Monokai it's less distinct, but I'd say let's roll with it, I'm definitely not ready to open the Pandora's box of testing the package across several different themes :)

![screen shot 2016-01-21 at 10 28 34](https://cloud.githubusercontent.com/assets/69962/12478445/df37c86a-c02c-11e5-9d53-6932e5fd09f2.png)

**Comment by hlvnst** [comment]
You're right, that *is* a Pandora's box. I did check a few themes, but Monokai wasn't one of them. I'll be sure to check it, too, in the future.

But since you're OK with it, I'll leave it as is for now. We can always change things up if people start complaining.

Also, I noticed you granted me write access to this repository. I take it this means you'd like me to push fixes and close issues directly instead of issuing PR's?
